### PR TITLE
Remove some unused sd functions

### PR DESCRIPTION
**Pull request by benma**

### CODE REVIEW

**Comment by benma** [comment]
sd_write() never even worked as the `bool replace` param is fed into f_open as the `uint8_t mode` with a wrong value

File: src/sd.c [hunk 1]
```diff
@@ -162,7 +162,7 @@ bool sd_write_bin(
     const uint16_t length,
     bool replace)
 {
-    if (!length || length > SD_MAX_FILE_SIZE) {
+    if (!strlens(fn) || data == NULL || !length || length > SD_MAX_FILE_SIZE) {
```
Comment (NickeZ) [inline src/sd.c:165]: Seems wrong to me to return false if length is zero. But that is the current behavior, so we can keep it.

**Review by NickeZ — CHANGES_REQUESTED** [review]
Could you also update the docs for `sd_write_bin`?

File: src/sd.c (hunk 1 above)
Comment (benma) [reply src/sd.c:165]: Done, PTAL

**Review by NickeZ — APPROVED** [review]
utACK

**Pull request merged**